File: vue-dashboard/js/backgroundClaude.js
// Initialize canvas
    const canvas = document.getElementById('particleCanvas');
    const ctx = canvas.getContext('2d');
    
    // Set canvas to full window size
    function resizeCanvas() {
      canvas.width = window.innerWidth;
      canvas.height = window.innerHeight;
    }
    
    // Track mouse position
    let mouse = {
      x: undefined,
      y: undefined,
      radius: 150
    };
    
    window.addEventListener('mousemove', function(event) {
      mouse.x = event.x;
      mouse.y = event.y;
    });
    
    // Create particle class
    class Particle {
      constructor(x, y, size, color, weight) {
        this.x = x;
        this.y = y;
        this.size = size;
        this.color = color;
        this.weight = weight;
        this.originalSize = size;
        this.originalX = x;
        this.originalY = y;
        this.connectionDistance = 200;
        this.mode = 'financial'; // default mode
      }
      
      update() {
        // Mouse repulsion (in both modes)
        if (mouse.x !== undefined && mouse.y !== undefined) {
          let dx = this.x - mouse.x;
          let dy = this.y - mouse.y;
          let distance = Math.sqrt(dx * dx + dy * dy);
          
          if (distance < mouse.radius) {
            const angle = Math.atan2(dy, dx);
            const force = (mouse.radius - distance) / mouse.radius;
            const pushX = Math.cos(angle) * force * 5;
            const pushY = Math.sin(angle) * force * 5;
            
            this.x += pushX;
            this.y += pushY;
          }
        }
        
        // Different movement logic based on mode
        if (this.mode === 'financial') {
          // Financial mode: circular motion like currency flow
          const time = Date.now() * 0.001;
          const radius = 30 + this.weight * 20;
          const speed = 0.3 - this.weight * 0.2;
          
          // Calculate new position based on circular motion
          const newX = this.originalX + Math.cos(time * speed + this.weight * 10) * radius;
          const newY = this.originalY + Math.sin(time * speed + this.weight * 10) * radius;
          
          // Smoothly interpolate to new position
          this.x += (newX - this.x) * 0.03;
          this.y += (newY - this.y) * 0.03;
        } else {
          // User data mode: gentle floating with slight gravitational pull
          if (Math.abs(this.x - this.originalX) > 100 || Math.abs(this.y - this.originalY) > 100) {
            this.x += (this.originalX - this.x) * 0.01;
            this.y += (this.originalY - this.y) * 0.01;
          } else {
            this.x += (Math.random() - 0.5) * 0.5;
            this.y += (Math.random() - 0.5) * 0.5;
          }
        }
        
        // Ensure particles stay within bounds
        if (this.x < 0) this.x = 0;
        if (this.x > canvas.width) this.x = canvas.width;
        if (this.y < 0) this.y = 0;
        if (this.y > canvas.height) this.y = canvas.height;
        
        // Draw the particle
        this.draw();
      }
      
      draw() {
        ctx.beginPath();
        ctx.arc(this.x, this.y, this.size, 0, Math.PI * 2);
        ctx.fillStyle = this.color;
        ctx.fill();
        
        // Add glow effect
        ctx.shadowColor = this.color;
        ctx.shadowBlur = 15;
        ctx.fill();
        ctx.shadowBlur = 0;
      }
      
      setMode(mode) {
        this.mode = mode;
        
        // Reset position when changing modes
        if (mode === 'financial') {
          this.color = getRandomFinancialColor();
        } else {
          this.color = getRandomUserColor();
        }
      }
    }
    
    // Color palettes
    function getRandomFinancialColor() {
      const colors = [
        'rgba(37, 248, 143, 0.7)',  // Green
        'rgba(22, 243, 232, 0.7)',  // Teal
        'rgba(95, 132, 252, 0.7)',  // Blue
        'rgba(255, 220, 100, 0.7)'  // Gold
      ];
      return colors[Math.floor(Math.random() * colors.length)];
    }
    
    function getRandomUserColor() {
      const colors = [
        'rgba(127, 255, 200, 0.7)',  // Soft green
        'rgba(200, 255, 255, 0.7)',  // Light cyan
        'rgba(180, 180, 255, 0.7)',  // Light blue
        'rgba(255, 200, 200, 0.7)'   // Light pink
      ];
      return colors[Math.floor(Math.random() * colors.length)];
    }
    
    // Initialize particles
    let particles = [];
    
    function initParticles() {
      particles = [];
      const particleCount = window.innerWidth < 768 ? 50 : 100;
      
      for (let i = 0; i < particleCount; i++) {
        const x = Math.random() * canvas.width;
        const y = Math.random() * canvas.height;
        const size = Math.random() * 2.5 + 1.5;
        const color = getRandomFinancialColor();
        const weight = Math.random(); // Used for movement calculations
        
        particles.push(new Particle(x, y, size, color, weight));
      }
    }
    
    // Animation loop
    function animate() {
      ctx.clearRect(0, 0, canvas.width, canvas.height);
      
      // Update and draw particles
      for (let i = 0; i < particles.length; i++) {
        particles[i].update();
      }
      
      // Draw connections between particles that are close to each other
      for (let i = 0; i < particles.length; i++) {
        for (let j = i + 1; j < particles.length; j++) {
          const dx = particles[i].x - particles[j].x;
          const dy = particles[i].y - particles[j].y;
          const distance = Math.sqrt(dx * dx + dy * dy);
          
          if (distance < particles[i].connectionDistance) {
            ctx.beginPath();
            ctx.strokeStyle = particles[i].color.replace('0.7', (1 - distance / particles[i].connectionDistance) * 0.5);
            ctx.lineWidth = 0.5;
            ctx.moveTo(particles[i].x, particles[i].y);
            ctx.lineTo(particles[j].x, particles[j].y);
            ctx.stroke();
          }
        }
      }
      
      requestAnimationFrame(animate);
    }
    
    // Mode switching functionality
    document.getElementById('financialMode').addEventListener('click', function() {
      particles.forEach(particle => particle.setMode('financial'));
    });
    
    document.getElementById('userMode').addEventListener('click', function() {
      particles.forEach(particle => particle.setMode('user'));
    });
    
    // Button interactions
    const buttons = document.querySelectorAll('button');
    buttons.forEach(button => {
      button.addEventListener('mouseenter', function() {
        mouse.radius = 100;
      });
      
      button.addEventListener('mouseleave', function() {
        mouse.radius = 150;
      });
      
      button.addEventListener('click', function() {
        if (this.id !== 'financialMode' && this.id !== 'userMode') {
          // Create explosion effect on button click
          for (let i = 0; i < 20; i++) {
            setTimeout(() => {
              const randomX = this.getBoundingClientRect().left + this.offsetWidth / 2 + (Math.random() - 0.5) * 100;
              const randomY = this.getBoundingClientRect().top + this.offsetHeight / 2 + (Math.random() - 0.5) * 100;
              
              // Create temporary particle for explosion effect
              const size = Math.random() * 3 + 2;
              const color = getRandomFinancialColor();
              const tempParticle = new Particle(randomX, randomY, size, color, Math.random());
              particles.push(tempParticle);
              
              // Remove temporary particle after animation
              setTimeout(() => {
                const index = particles.indexOf(tempParticle);
                if (index > -1) {
                  particles.splice(index, 1);
                }
              }, 2000);
            }, i * 50);
          }
        }
      });
    });
    
    // Add simulated data update periodically
    setInterval(() => {
      const statValues = document.querySelectorAll('.stat-value');
      const statChanges = document.querySelectorAll('.stat-change');
      
      statValues.forEach((statValue, index) => {
        if (index === 0) {
          // Total Users
          const currentUsers = parseInt(statValue.textContent);
          const change = Math.floor(Math.random() * 5) - 1;
          statValue.textContent = currentUsers + change;
          
          const changePercent = ((change / currentUsers) * 100).toFixed(1);
          statChanges[index].textContent = `${changePercent > 0 ? '+' : ''}${changePercent}% from last month`;
          statChanges[index].className = 'stat-change' + (changePercent < 0 ? ' negative' : '');
        }
        
        // Similar updates for other stats...
      });
    }, 10000);
    
    // Initialize everything
    window.addEventListener('resize', resizeCanvas);
    resizeCanvas();
    initParticles();
    animate();

    
    
async function fetchDashboardData() {
    try {
      const res = await fetch(`${API_BASE}/dashboard-stats`);
      const stats = await res.json();
      
      // Update the UI with real data
      document.querySelector('.stat-card:nth-child(1) .stat-value').textContent = stats.totalUsers;
      document.querySelector('.stat-card:nth-child(2) .stat-value').textContent = stats.totalAccounts;
      // etc.
    } catch (err) {
      console.error('Error fetching dashboard data:', err);
    }
  }
  
  // Call this function when page loads
  fetchDashboardData();

File: vue-dashboard/js/backgroundClaude.py
import math
import random
import js
from pyodide.ffi import create_proxy
from pyodide import create_once_callable
import asyncio

# Initialize canvas
canvas = js.document.getElementById('particleCanvas')
ctx = canvas.getContext('2d')

# Set canvas to full window size
def resizeCanvas():
    canvas.width = js.window.innerWidth
    canvas.height = js.window.innerHeight

# Track mouse position
mouse = {'x': None, 'y': None, 'radius': 150}

def on_mousemove(event):
    mouse['x'] = event.x
    mouse['y'] = event.y

js.window.addEventListener('mousemove', create_proxy(on_mousemove))

# Particle class
class Particle:
    def __init__(self, x, y, size, color, weight):
        self.x = x
        self.y = y
        self.size = size
        self.color = color
        self.weight = weight
        self.originalSize = size
        self.originalX = x
        self.originalY = y
        self.connectionDistance = 200
        self.mode = 'financial'

    def update(self):
        # Mouse repulsion
        if mouse['x'] is not None and mouse['y'] is not None:
            dx = self.x - mouse['x']
            dy = self.y - mouse['y']
            distance = math.sqrt(dx * dx + dy * dy)

            if distance < mouse['radius']:
                angle = math.atan2(dy, dx)
                force = (mouse['radius'] - distance) / mouse['radius']
                pushX = math.cos(angle) * force * 5
                pushY = math.sin(angle) * force * 5
                self.x += pushX
                self.y += pushY

        # Movement logic based on mode
        if self.mode == 'financial':
            time = js.Date.now() * 0.001
            radius = 30 + self.weight * 20
            speed = 0.3 - self.weight * 0.2
            newX = self.originalX + math.cos(time * speed + self.weight * 10) * radius
            newY = self.originalY + math.sin(time * speed + self.weight * 10) * radius
            self.x += (newX - self.x) * 0.03
            self.y += (newY - self.y) * 0.03
        else:
            if abs(self.x - self.originalX) > 100 or abs(self.y - self.originalY) > 100:
                self.x += (self.originalX - self.x) * 0.01
                self.y += (self.originalY - self.y) * 0.01
            else:
                self.x += (random.random() - 0.5) * 0.5
                self.y += (random.random() - 0.5) * 0.5

        # Keep particles in bounds
        self.x = max(0, min(self.x, canvas.width))
        self.y = max(0, min(self.y, canvas.height))

        self.draw()

    def draw(self):
        ctx.beginPath()
        ctx.arc(self.x, self.y, self.size, 0, math.pi * 2)
        ctx.fillStyle = self.color
        ctx.fill()
        ctx.shadowColor = self.color
        ctx.shadowBlur = 15
        ctx.fill()
        ctx.shadowBlur = 0

    def setMode(self, mode):
        self.mode = mode
        self.color = getRandomFinancialColor() if mode == 'financial' else getRandomUserColor()

# Color palettes
def getRandomFinancialColor():
    colors = [
        'rgba(37, 248, 143, 0.7)',  # Green
        'rgba(22, 243, 232, 0.7)',  # Teal
        'rgba(95, 132, 252, 0.7)',  # Blue
        'rgba(255, 220, 100, 0.7)'  # Gold
    ]
    return random.choice(colors)

def getRandomUserColor():
    colors = [
        'rgba(127, 255, 200, 0.7)',  # Soft green
        'rgba(200, 255, 255, 0.7)',  # Light cyan
        'rgba(180, 180, 255, 0.7)',  # Light blue
        'rgba(255, 200, 200, 0.7)'   # Light pink
    ]
    return random.choice(colors)

# Initialize particles
particles = []

def initParticles():
    global particles
    particles = []
    particleCount = 50 if js.window.innerWidth < 768 else 100
    for _ in range(particleCount):
        x = random.random() * canvas.width
        y = random.random() * canvas.height
        size = random.random() * 2.5 + 1.5
        color = getRandomFinancialColor()
        weight = random.random()
        particles.append(Particle(x, y, size, color, weight))

# Animation loop
async def animate():
    ctx.clearRect(0, 0, canvas.width, canvas.height)
    
    for particle in particles:
        particle.update()

    # Draw connections
    for i, p1 in enumerate(particles):
        for p2 in particles[i + 1:]:
            dx = p1.x - p2.x
            dy = p1.y - p2.y
            distance = math.sqrt(dx * dx + dy * dy)
            if distance < p1.connectionDistance:
                ctx.beginPath()
                opacity = (1 - distance / p1.connectionDistance) * 0.5
                ctx.strokeStyle = p1.color.replace('0.7', str(opacity))
                ctx.lineWidth = 0.5
                ctx.moveTo(p1.x, p1.y)
                ctx.lineTo(p2.x, p2.y)
                ctx.stroke()

    # Schedule next frame
    js.window.requestAnimationFrame(create_once_callable(lambda _: asyncio.ensure_future(animate())))

# Mode switching
def setFinancialMode(_):
    for particle in particles:
        particle.setMode('financial')

def setUserMode(_):
    for particle in particles:
        particle.setMode('user')

js.document.getElementById('financialMode').addEventListener('click', create_proxy(setFinancialMode))
js.document.getElementById('userMode').addEventListener('click', create_proxy(setUserMode))

# Button interactions
def on_button_mouseenter(event):
    mouse['radius'] = 100

def on_button_mouseleave(event):
    mouse['radius'] = 150

async def on_button_click(event):
    button = event.target
    if button.id not in ['financialMode', 'userMode']:
        for i in range(20):
            await asyncio.sleep(i * 0.05)
            rect = button.getBoundingClientRect()
            randomX = rect.left + rect.width / 2 + (random.random() - 0.5) * 100
            randomY = rect.top + rect.height / 2 + (random.random() - 0.5) * 100
            size = random.random() * 3 + 2
            color = getRandomFinancialColor()
            tempParticle = Particle(randomX, randomY, size, color, random.random())
            particles.append(tempParticle)
            js.setTimeout(create_once_callable(lambda: particles.remove(tempParticle) if tempParticle in particles else None), 2000)

buttons = js.document.querySelectorAll('button')
for button in buttons:
    button.addEventListener('mouseenter', create_proxy(on_button_mouseenter))
    button.addEventListener('mouseleave', create_proxy(on_button_mouseleave))
    button.addEventListener('click', create_proxy(lambda e: asyncio.ensure_future(on_button_click(e))))

# Initialize everything
def on_resize(_):
    resizeCanvas()
    initParticles()

js.window.addEventListener('resize', create_proxy(on_resize))
resizeCanvas()
initParticles()
asyncio.ensure_future(animate())